### PR TITLE
feat: add linting documentation

### PR DESCRIPTION
**Pull request by matthew-chirgwin**
- adds first version of linting documentation, along with proposed tools
to use, as well as some suggested rules to follow for UI code

Signed-off-by: Matthew Chirgwin <chirmatt@uk.ibm.com>

### CODE REVIEW

File: docs/Linting.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+This section will detail the rules implemented across the codebase, and a rationale to why these rules have been chosen. Apart from the [Git commit checks](#Git-commit-checks), all linting will occur pre commit, via a Husky hook. Alternatively, a developer can run these pre commit checks by running:
+
+```
+npm run lint
+```
+
+to run all checks, or for indvidual lint checks, the command mentioned in each section respectivley.
+
+To pass linting, no errors or warnings are allowed - ie all rules must be satisfied. It is possible to override and ignore rules, but this should only be done as a last resort, with an accompanying explination.
```
Comment (JQrdan) [inline docs/Linting.md]: typo on 'explination'

**Review by JQrdan — CHANGES_REQUESTED** [review]
Tooling looks good! Happy to see a CSS linter on there too

File: docs/Linting.md (hunk 1 above)
Comment (matthew-chirgwin) [reply docs/Linting.md]: Thanks Jordan! Good spot, will fix now 👍

**Comment by JQrdan** [comment]
Is there a way to hook in the style lint with prettier in the same way that eslint can be handled by it? I'm sure there's a stylelint plugin that can auto-fix spacing but if we have any more specific style requirements it could be handy for that to all be handled by prettier.

**Comment by matthew-chirgwin** [comment]
> Is there a way to hook in the style lint with prettier in the same way that eslint can be handled by it? I'm sure there's a stylelint plugin that can auto-fix spacing but if we have any more specific style requirements it could be handy for that to all be handled by prettier.

I absolutley agree. My thought was to divide and conquer linting responsibilities between a tool for syntax (ie eslint, stylelint) and one for formatting (prettier).

**Review by nictownsend — APPROVED** [review]
LGTM

**Review by dlabaj — APPROVED** [review]
Would like test this out with typescript this week.  Was told recently that eslint can be used with typescript looking into that.  Also agree that having prettier as well would be great. 

LGTM

File: docs/Linting.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+To pass linting, no errors or warnings are allowed - ie all rules must be satisfied. It is possible to override and ignore rules, but this should only be done as a last resort, with an accompanying explanation.
+
+### All files
+
+The following rules are applied to all files:
+
+| Rule/Ruleset                                                                              |                                       Rationale |                                                         Implemented by | Auto fixes? |
+| ----------------------------------------------------------------------------------------- | ----------------------------------------------: | ---------------------------------------------------------------------: | ----------: |
+| Must have the Strimzi copyright header (in the appropriate format) at the top of the file | Requirement for all code contributed to Strimzi | [license-check-and-add](https://github.com/awjh/license-check-and-add) |         Yes |
```
Comment (ppatierno) [inline docs/Linting.md:46]: where is the copyright header format defined? are you going to use the same we have in the Java source code?

**Review by ppatierno — APPROVED** [review]
Happy with this. I just asked a question.

File: docs/Linting.md (hunk 2 above)
Comment (matthew-chirgwin) [reply docs/Linting.md:46]: @samuel-hawker kindly provided me with this:
```
Copyright Strimzi authors.
License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
```
I would have this defined in a file, which the tooling then inserts as required (with the right comments)

**Review by scholzj — APPROVED** [review]

**Comment by scholzj** [comment]
@JQrdan Are you fine with this? Looks like you requested some changes on this PR.

**Pull request merged**